api/common/sharing/utils.go

```diff
@@ -37,13 +37,8 @@ func ExtractReferencedInstanceID(req *web.Request, repository storage.Repository
 			return "", err
 		}
 	} else {
-		sharedInstances, err := getSharedInstances(ctx, repository, referencePlan.ServiceOfferingID, tenantIdentifier, getTenantId())
-		if err != nil {
-			log.C(ctx).Errorf("Failed to retrieve shared instances: %s", err)
-			return "", err
-		}
-
-		referencedInstance, err = filterInstancesBySelectors(ctx, repository, sharedInstances, parameters, smaap, referencePlan.ServiceOfferingID)
+		selectors := parameters[instance_sharing.SelectorsKey].Map()
```

Comment:
see the comment belowm the parameters[instance_sharing.SelectorsKey] could be null, you return null in validateParameters function

Reply:
can't be null, as explained there in the comment.

Reply:
ok